updateData.js

```diff
@@ -0,0 +1,132 @@
+        })
+    }
+
+    for (let sessionId in sessions) {
+        sessions[sessionId].speakers = sessions[sessionId].speakers.map(speakerId => {
+            return {id: speakerId, slug: speakers[speakerId].slug}
+        })
+    }
+
+    writeDataFile('sessions.json', sessions);
+    writeDataFile('speakers.json', speakers);
+    writeDataFile('rooms.json', sessionize.rooms);
+}
+
+
+function parseCategories(categories) {
+    var levels = {};
+    var formats = {};
+
+    for (let category of categories) {
+        if (category.title == 'Level') {
+            for (level of category.items) {
+                levels[level.id] = level;
+            }
+        } else if (category.title == 'Session format') {
+            for (format of category.items) {
+                formats[format.id] = format;
+            }
+        }
+    }
+
+    return [levels, formats]
+}
+
+function buildSpeakers(speakersData) {
+    for (let speaker of speakersData) {
+
+        // build full links
+        for (let link of speaker.links) {
+            link.name = link.title;
+            switch (link.linkType) {
+                case 'Twitter':
+                    link.name = '@' + link.url.replace(/https*:\/\/(www\.)*twitter.com\//gi, '')
+                                              .replace(/\/?(\?.*)?$/, '');
+                    break;
+                case 'Blog':
+                case 'Company_Website':
+                    link.name = link.url.replace(/https*:\/\/(www\.)*/gi, '')
+                                        .replace(/\/?(\?.*)?$/, '')
+                                        .replace(/\/.*/, '');
+                    break;       
+            }
+        }
+
+        // create slug
+        speaker.slug = slugify(speaker.firstName + " " + speaker.lastName)
+    }
+
+    return flattenArrayToObj(speakersData)
+}
+
+
+function buildSessions(sessionsData, levels, formats) {
+    for (let session of sessionsData) {
+
+        // apply level and category labels
+        for (let categoryId of session.categoryItems) {
+            if (categoryId in levels) {
+                session.level = levels[categoryId].name;
+            } else if (categoryId in formats) {
+                session.format = formats[categoryId].name;
+            }
+        }
+
+        // create slug
+        session.slug = slugify(session.title);
+    }
+
+    return flattenArrayToObj(sessionsData);
+}
+
+
+
+function writeDataFile(filename, object) {
+    let projectRoot = path.normalize(__dirname);
+
+    let filePath = `${projectRoot}/src/_data/${filename}`;
+    let content = JSON.stringify(object, null, 4);
+
+    fs.writeFile(filePath, content, function(err) {
+        if(err) { return console.log(err); }
+        console.log(`Sessionize data written to ${filePath}`);
+    });
+}
+
+function flattenArrayToObj(array) {
+    let object = {};
+
+    for (let item of array) {
+        object[item.id] = item;
+    }
+
+    return object;
+}
+
+function slugify(s) {
```

Comment:
Another discussion topic.  We used to have straight IDs for slugs for speakers and sessions, but those IDs look a little long and ugly:

```none
https://www.vtcodecamp.org/sessions/#124040
https://www.vtcodecamp.org/speakers/#cb70d1bd-13ab-4a16-abfd-cf48b37ec30d
```

Since we're building slugs from speaker names and session titles, and those values are allowed to change, it's not guaranteed to be a permanent lifetime indicator like ids.  But after we winterize the site, it'll stay static just fine.  And in the meantime, they'll still land on the correct page if the value changes, and it'll build a lot friendlier urls

Reply:
I don't feel strongly about this either way.  I'm fine trading ugly IDs for slightly less complexity.  Somehow I guess I don't consider human readability to be as important for a url fragment as for the main part of the url.  However, the slugs built from the names *do* look nicer.  I'm not worried about the values changing either.  A wrong url fragment just doesn't scroll the right part of the page.  It won't give a user a 404 error.  The only suggestion I have is that I think the slugs would look better all lowercase.

Reply:
It's really the longer user GUIDs that really made me consider using display names.  I agree that lowercasing the slugs looks a little cleaner.